taranis/allele_calling.py

```diff
@@ -23,105 +25,142 @@
+                # allele is labled as ASM
+                pdb.set_trace()
+                return ["ASM", allele_name, allele_details]
+            # check if contig is longer than allele
+            if int(column_blast_res[3]) < int(column_blast_res[4]):
+                # allele is labled as ALM
+                pdb.set_trace()
+                return ["ALM", allele_name, allele_details]
+            if int(column_blast_res[3]) == int(column_blast_res[4]):
+                # allele is labled as INF
+                pdb.set_trace()
+                return ["INF", allele_name, allele_details]
+
         pdb.set_trace()
 
-    def analyze_sample(self):
+    def search_match_allele(self):
         # Create  blast db with sample file
-        self.sample_blast = taranis.blast.Blast("nucl")
-        _ = self.sample_blast.create_blastdb(self.sample_file, self.blast_dir)
-        result = {}
+
+        result = {"allele_type": {}, "allele_match": {}, "allele_details": {}}
         # pdb.set_trace()
         for ref_allele in self.ref_alleles:
             # schema_alleles = os.path.join(self.schema, ref_allele)
             # parallel in all CPUs in cluster node
-            result[ref_allele] = self.search_alleles(ref_allele)
+            alleles = OrderedDict()
+            match_found = False
+            with open(ref_allele, "r") as fh:
+                for record in SeqIO.parse(fh, "fasta"):
+                    alleles[record.id] = str(record.seq)
 
-        pdb.set_trace()
-        return
+            for r_id, r_seq in alleles.items():
+                # create file in memory to increase speed
+                query_file = io.StringIO()
+                query_file.write(">" + r_id + "\n" + r_seq)
+                query_file.seek(0)
+                blast_result = self.blast_obj.run_blast(
+                    query_file.read(), perc_identity=90, query_type="stdin"
+                )
+                if len(blast_result) > 0:
+                    match_found = True
+                    break
+            # Close object and discard memory buffer
+            query_file.close()
+            if match_found:
+                allele_file = os.path.join(self.schema, os.path.basename(ref_allele))
+                # blast_result = self.blast_obj.run_blast(q_file,perc_identity=100)
+                allele_name = Path(allele_file).stem
+                # pdb.set_trace()
+                (
+                    result["allele_type"][allele_name],
+                    result["allele_match"][allele_name],
+                    result["allele_details"][allele_name],
+                ) = self.assign_allele_type(blast_result, allele_file, allele_name)
+                # pdb.set_trace()
+            else:
+                # Sample does not have a reference allele to be matched
+                # Keep LNF info
+                # ver el codigo de espe
+                # lnf_tpr_tag()
+                pdb.set_trace()
```

Comment:
another pdb, i guess this is testing you are doing now

Reply:
removed